docs/javascript/object-constructor.md

```diff
@@ -0,0 +1,281 @@
+
+```javascript
+function Person(first, last, age, eyecolor) {
+  this.firstName = first;
+  this.lastName = last;
+  this.age = age;
+  this.eyeColor = eyecolor;
+  this.nationality = "English";
+}
+```
+
+### Adding a Property to an Object
+
+Adding a property to a created object is easy:
+
+```javascript
+myFather.nationality = "English";
+```
+
+:::info Note
+
+> The property will be added to `myFather`. Not to any other Person objects.
+
+:::
+### Adding a Property to a Constructor
+
+You cannot add a new property directly to an object constructor. To add a new property, you must add it to the constructor function prototype:
+
+```javascript
+Person.prototype.nationality = "English";
+```
+
+## Adding a Method to a Constructor
+
+You cannot add a new method directly to an object constructor function. This code will produce a TypeError:
+
+```javascript
+Person.changeName = function (name) {
+  this.lastName = name;
+}
+
+myMother.changeName("Doe");
+// TypeError: myMother.changeName is not a function
+```
+
+## Built-in JavaScript Constructors
+
+JavaScript has built-in constructors for all native objects:
+
+```javascript
+new Object();   // A new Object object
+new Array();    // A new Array object
+new Map();      // A new Map object
+new Set();      // A new Set object
+new Date();     // A new Date object
+new RegExp();   // A new RegExp object
+new Function(); // A new Function object
+```
+
+## Best Practices
+
+- Use object literals `{}` instead of `new Object()`.
+- Use array literals `[]` instead of `new Array()`.
+- Use pattern literals `/()/` instead of `new RegExp()`.
+- Use function expressions `() {}` instead of `new Function()`.
+
+### Examples
+
+```javascript
+"";           // primitive string
+0;            // primitive number
+false;        // primitive boolean
+
+{};           // object object
+[];           // array object
+/()/;         // regexp object
+function(){}; // function
+```
+
+### Creating Another Person Object
+
+```javascript
+const mySelf = new Person("Johnny", "Rally", 22, "green");
+```
+
+![JavaScript Constructors](https://miro.medium.com/v2/resize:fit:804/1*oREMGa3m7L4j1DzdNJc6GA.jpeg)
+
+## Constructor Functions in Detail
+
+### Constructor
+
+A constructor is a function that initializes an object. In JavaScript, constructors are more similar to normal Java constructors.
+
+### Object Constructor
+
+In JavaScript, there is a special constructor function known as `Object()` that is used to create and initialize an object. The return value of the `Object()` constructor is assigned to a variable. The variable contains a reference to the new object. We need an object constructor to create an object “type” that can be used multiple times without redefining the object every time.
+
+#### Example
+
+```javascript
+function GFG(A, B, C) {
+    this.g = A;
+    this.f = B;
+    this.gg = C;
+}
+```
+Here, `GFG` is the constructor name and `A`, `B`, `C` are the arguments of the constructor.
```

Comment:
Please modify the example so that it does not use any content from the existing site. We need to create something unique for our use.